Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php

```diff
@@ -179,6 +179,11 @@ private function handlePublishWorkspace(
     ): \Generator {
         $workspace = $this->requireWorkspace($command->workspaceName, $commandHandlingDependencies);
         $baseWorkspace = $this->requireBaseWorkspace($workspace, $commandHandlingDependencies);
+        if (!$workspace->hasPublishableChanges()) {
+            // no-op
+            return;
```

Comment:
I would suggest to throw a dedicated exception in this case to make it really explicit.
We can always decide to catch it in the application (i.e. Neos) if we need to but just ignoring that case is "dangerous" IMO

Reply:
@mhsdesign promised to tackle this in a follow-up :)